projects/step-frontend/src/lib/modules/execution/components/execution-legacy-switcher/execution-legacy-switcher.component.html

```diff
@@ -0,0 +1,17 @@
+@if (!forceLegacyReporting()) {
+  <step-btn-group class="new-execution-report">
+    <mat-slide-toggle
+      [formControl]="toggleControl"
+      color="primary"
+      name="new-execution-report"
+      [disabled]="isDisabled()"
+      [matTooltip]="
+        isDisabled()
+          ? _executionViewModeService.getNewExecutionDeactivatedReason(execution()!)
+          : 'Select if you prefer the legacy or the new execution view. (stored in the browsers local storage)'
+      "
```

Comment:
Function inside template.
In this component it can be avoided, if the logic will be moved to signal
```
protected tooltipMessage = computed(() => {
   const isDisabled = this.isDisabled();
   const execution = this.execution();
   if (!execution) {
       return '';
   }
   return isDisabled ? 
           this._executionViewModeService.getNewExecutionDeactivatedReason(execution) :
           'Select if you prefer the legacy or the new execution view. (stored in the browsers local storage)';
});
```